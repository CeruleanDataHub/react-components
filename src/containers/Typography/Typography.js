import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

const sizes = { normal: "16px", large: "40px" };
const colors = {
  red: "#ff0066",
  green: "#90ee7e",
  blue: "#1ea7fd",
  white: "white",
  black: "black",
  gray: "#999999"
};

const fontFamilies = { exo: '"Exo 2"', normal: '"Open Sans"' };

const Container = styled.div`
  font-family: ${({ exo }) =>
    `${exo ? fontFamilies.exo : fontFamilies.normal}, sans-serif`};
  font-size: ${({ size }) => sizes[size] || sizes.normal};
  color: ${({ textColor }) => colors[textColor] || colors.black};
`;

export const Typography = ({ size, color, exo = false, children }) => (
  <Container textColor={color} size={size} exo={exo}>
    {children}
  </Container>
);

Typography.propTypes = {
  /** Text color, values predefined */
  color: PropTypes.oneOf(["red", "green", "blue", "white", "black", "gray"]),
  /** Text size, values predefined */
  size: PropTypes.oneOf(["normal", "large"]),
  /** Define whether to render text with Exo 2 font or Open Sans  */
  exo: PropTypes.bool,
  /** React node */
  children: PropTypes.node
};

Typography.defaultProps = {
  color: "black",
  size: "normal",
  exo: false,
  children: ""
};
