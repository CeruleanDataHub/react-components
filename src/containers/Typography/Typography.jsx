import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

const sizes = { normal: "16px", large: "40px" };
const colors = {
  red: "#ff0066",
  green: "#90ee7e",
  blue: "#1ea7fd",
  white: "white",
  black: "black",
  gray: "#999999"
};

const fontFamilies = { exo: '"Exo 2"', openSans: '"Open Sans"' };

const fontFamilyStyle = fontFamily => {
  if (!fontFamily) {
    return "unset";
  }
  return `${fontFamilies[fontFamily]}, sans-serif`;
};

const Container = styled.div`
  font-family: ${({ fontFamily }) => fontFamilyStyle(fontFamily)};
  font-size: ${({ size }) => sizes[size] || sizes.normal};
  color: ${({ textColor }) => colors[textColor] || "unset"};
`;

export const Typography = ({ size, color, fontFamily, children }) => (
  <Container textColor={color} size={size} fontFamily={fontFamily}>
    {children}
  </Container>
);

Typography.propTypes = {
  /** Text color, values predefined */
  color: PropTypes.oneOf([
    "",
    "red",
    "green",
    "blue",
    "white",
    "black",
    "gray"
  ]),
  /** Text size, values predefined */
  size: PropTypes.oneOf(["normal", "large"]),
  /** Define whether to render text with Exo 2 font or Open Sans. If not defined, renders text with default font */
  fontFamily: PropTypes.oneOf(["exo", "openSans"]),
  /** React node */
  children: PropTypes.node
};

Typography.defaultProps = {
  color: "",
  size: "normal",
  fontFamily: null,
  children: ""
};
