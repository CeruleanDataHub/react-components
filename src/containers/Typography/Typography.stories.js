import React from "react";

import { Typography } from "./Typography";

export default {
  title: "Typography",
  component: Typography
};

export const TypographyStory = () => <Typography>Typography</Typography>;

export const TypographyBlackStory = () => (
  <Typography color="black">Typography</Typography>
);

export const TypographyGrayStory = () => (
  <Typography color="gray">Typography</Typography>
);

export const TypographyWhiteStory = () => (
  <div style={{ backgroundColor: "black" }}>
    <Typography color="white">Typography</Typography>
  </div>
);

export const TypographyRedStory = () => (
  <Typography color="red">Typography</Typography>
);

export const TypographyGreenStory = () => (
  <Typography color="green">Typography</Typography>
);

export const TypographyBlueStory = () => (
  <Typography color="blue">Typography</Typography>
);

export const TypographyNormalStory = () => (
  <Typography size="normal">Typography</Typography>
);

export const TypographyLargeStory = () => (
  <Typography size="large">Typography</Typography>
);

export const TypographyNormalBlackStory = () => (
  <Typography size="normal" color="black">
    Typography
  </Typography>
);

export const TypographyNormalGrayStory = () => (
  <Typography size="normal" color="gray">
    Typography
  </Typography>
);

export const TypographyNormalWhiteStory = () => (
  <div style={{ backgroundColor: "black" }}>
    <Typography size="normal" color="white">
      Typography
    </Typography>
  </div>
);

export const TypographyNormalRedStory = () => (
  <Typography size="normal" color="red">
    Typography
  </Typography>
);

export const TypographyNormalGreenStory = () => (
  <Typography size="normal" color="green">
    Typography
  </Typography>
);

export const TypographyNormalBlueStory = () => (
  <Typography size="normal" color="blue">
    Typography
  </Typography>
);

export const TypographyLargeBlackStory = () => (
  <Typography size="large" color="black">
    Typography
  </Typography>
);

export const TypographyLargeGrayStory = () => (
  <Typography size="large" color="gray">
    Typography
  </Typography>
);

export const TypographyLargeWhiteStory = () => (
  <div style={{ backgroundColor: "black" }}>
    <Typography size="large" color="white">
      Typography
    </Typography>
  </div>
);

export const TypographyLargeRedStory = () => (
  <Typography size="large" color="red">
    Typography
  </Typography>
);

export const TypographyLargeGreenStory = () => (
  <Typography size="large" color="green">
    Typography
  </Typography>
);

export const TypographyLargeBlueStory = () => (
  <Typography size="large" color="blue">
    Typography
  </Typography>
);

export const TypographyExoBlackStory = () => (
  <Typography color="black" exo>
    Typography
  </Typography>
);
export const TypographyExoGrayStory = () => (
  <Typography color="gray" exo>
    Typography
  </Typography>
);

export const TypographyExoWhiteStory = () => (
  <div style={{ backgroundColor: "black" }}>
    <Typography color="white" exo>
      Typography
    </Typography>
  </div>
);

export const TypographyExoRedStory = () => (
  <Typography color="red" exo>
    Typography
  </Typography>
);

export const TypographyExoGreenStory = () => (
  <Typography color="green" exo>
    Typography
  </Typography>
);

export const TypographyExoBlueStory = () => (
  <Typography color="blue" exo>
    Typography
  </Typography>
);

export const TypographyExoNormalStory = () => (
  <Typography size="normal" exo>
    Typography
  </Typography>
);

export const TypographyExoNormalBlackStory = () => (
  <Typography size="normal" color="black" exo>
    Typography
  </Typography>
);

export const TypographyExoNormalGrayStory = () => (
  <Typography size="normal" color="gray" exo>
    Typography
  </Typography>
);

export const TypographyExoNormalWhiteStory = () => (
  <div style={{ backgroundColor: "black" }}>
    <Typography size="normal" color="white" exo>
      Typography
    </Typography>
  </div>
);

export const TypographyExoNormalRedStory = () => (
  <Typography size="normal" color="red" exo>
    Typography
  </Typography>
);

export const TypographyExoNormalGreenStory = () => (
  <Typography size="normal" color="green" exo>
    Typography
  </Typography>
);

export const TypographyExoNormalBlueStory = () => (
  <Typography size="normal" color="blue" exo>
    Typography
  </Typography>
);

export const TypographyExoLargeStory = () => (
  <Typography size="large" exo>
    Typography
  </Typography>
);

export const TypographyExoLargeBlackStory = () => (
  <Typography size="large" color="black" exo>
    Typography
  </Typography>
);

export const TypographyExoLargeGrayStory = () => (
  <Typography size="large" color="gray" exo>
    Typography
  </Typography>
);

export const TypographyExoLargeWhiteStory = () => (
  <div style={{ backgroundColor: "black" }}>
    <Typography size="large" color="white" exo>
      Typography
    </Typography>
  </div>
);

export const TypographyExoLargeRedStory = () => (
  <Typography size="large" color="red" exo>
    Typography
  </Typography>
);

export const TypographyExoLargeGreenStory = () => (
  <Typography size="large" color="green" exo>
    Typography
  </Typography>
);

export const TypographyExoLargeBlueStory = () => (
  <Typography size="large" color="blue" exo>
    Typography
  </Typography>
);
