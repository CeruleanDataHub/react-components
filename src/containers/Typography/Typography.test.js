import React from "react";
import renderer from "react-test-renderer";

import { Typography } from "./Typography";

const colors = ["red", "green", "blue", "white", "black", "gray"];
const sizes = ["normal", "large"];
const fontFamilies = ["exo", "openSans"];

describe("Typography", () => {
  it("should render", () => {
    const component = renderer.create(<Typography>Test</Typography>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  fontFamilies.forEach(fontFamily => {
    describe(fontFamily, () => {
      it("should render font", () => {
        const component = renderer.create(
          <Typography fontFamily={fontFamily}>Test</Typography>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      describe("Colors", () => {
        colors.forEach(color => {
          it(`should render ${color} exo font`, () => {
            const component = renderer.create(
              <Typography color={color} fontFamily="exo">
                Test
              </Typography>
            );
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
          });
        });
      });

      describe("Sizes", () => {
        sizes.forEach(size => {
          it(`should render ${size} open sans font`, () => {
            const component = renderer.create(
              <Typography size={size} fontFamily="openSans">
                Test
              </Typography>
            );
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
          });
        });
      });

      describe("Sizes & Colors combinations", () => {
        colors.forEach(color => {
          sizes.forEach(size => {
            it(`should render ${color} ${size} exo font`, () => {
              const component = renderer.create(
                <Typography color={color} size={size} fontFamily="exo">
                  Test
                </Typography>
              );
              const tree = component.toJSON();
              expect(tree).toMatchSnapshot();
            });
          });
        });
      });
    });
  });
});
