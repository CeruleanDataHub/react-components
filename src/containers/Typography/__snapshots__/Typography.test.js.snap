// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography exo Colors should render black exo font 1`] = `
<div
  className="sc-AxjAm kBatYO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Colors should render blue exo font 1`] = `
<div
  className="sc-AxjAm eTPBNi"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Colors should render gray exo font 1`] = `
<div
  className="sc-AxjAm dfa-dQm"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Colors should render green exo font 1`] = `
<div
  className="sc-AxjAm kxpAJd"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Colors should render red exo font 1`] = `
<div
  className="sc-AxjAm hTKtVO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Colors should render white exo font 1`] = `
<div
  className="sc-AxjAm jGnDVe"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render black large exo font 1`] = `
<div
  className="sc-AxjAm cowTHD"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render black normal exo font 1`] = `
<div
  className="sc-AxjAm kBatYO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render blue large exo font 1`] = `
<div
  className="sc-AxjAm ebHQZj"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render blue normal exo font 1`] = `
<div
  className="sc-AxjAm eTPBNi"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render gray large exo font 1`] = `
<div
  className="sc-AxjAm bpqeif"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render gray normal exo font 1`] = `
<div
  className="sc-AxjAm dfa-dQm"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render green large exo font 1`] = `
<div
  className="sc-AxjAm cUMHOU"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render green normal exo font 1`] = `
<div
  className="sc-AxjAm kxpAJd"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render red large exo font 1`] = `
<div
  className="sc-AxjAm fOTfxj"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render red normal exo font 1`] = `
<div
  className="sc-AxjAm hTKtVO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render white large exo font 1`] = `
<div
  className="sc-AxjAm hWtwtf"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes & Colors combinations should render white normal exo font 1`] = `
<div
  className="sc-AxjAm jGnDVe"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo Sizes should render large open sans font 1`] = `
<div
  className="sc-AxjAm fCMKPq"
  fontFamily="openSans"
  size="large"
>
  Test
</div>
`;

exports[`Typography exo Sizes should render normal open sans font 1`] = `
<div
  className="sc-AxjAm bYePpP"
  fontFamily="openSans"
  size="normal"
>
  Test
</div>
`;

exports[`Typography exo should render font 1`] = `
<div
  className="sc-AxjAm MwpTk"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Colors should render black exo font 1`] = `
<div
  className="sc-AxjAm kBatYO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Colors should render blue exo font 1`] = `
<div
  className="sc-AxjAm eTPBNi"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Colors should render gray exo font 1`] = `
<div
  className="sc-AxjAm dfa-dQm"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Colors should render green exo font 1`] = `
<div
  className="sc-AxjAm kxpAJd"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Colors should render red exo font 1`] = `
<div
  className="sc-AxjAm hTKtVO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Colors should render white exo font 1`] = `
<div
  className="sc-AxjAm jGnDVe"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render black large exo font 1`] = `
<div
  className="sc-AxjAm cowTHD"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render black normal exo font 1`] = `
<div
  className="sc-AxjAm kBatYO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render blue large exo font 1`] = `
<div
  className="sc-AxjAm ebHQZj"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render blue normal exo font 1`] = `
<div
  className="sc-AxjAm eTPBNi"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render gray large exo font 1`] = `
<div
  className="sc-AxjAm bpqeif"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render gray normal exo font 1`] = `
<div
  className="sc-AxjAm dfa-dQm"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render green large exo font 1`] = `
<div
  className="sc-AxjAm cUMHOU"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render green normal exo font 1`] = `
<div
  className="sc-AxjAm kxpAJd"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render red large exo font 1`] = `
<div
  className="sc-AxjAm fOTfxj"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render red normal exo font 1`] = `
<div
  className="sc-AxjAm hTKtVO"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render white large exo font 1`] = `
<div
  className="sc-AxjAm hWtwtf"
  fontFamily="exo"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes & Colors combinations should render white normal exo font 1`] = `
<div
  className="sc-AxjAm jGnDVe"
  fontFamily="exo"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans Sizes should render large open sans font 1`] = `
<div
  className="sc-AxjAm fCMKPq"
  fontFamily="openSans"
  size="large"
>
  Test
</div>
`;

exports[`Typography openSans Sizes should render normal open sans font 1`] = `
<div
  className="sc-AxjAm bYePpP"
  fontFamily="openSans"
  size="normal"
>
  Test
</div>
`;

exports[`Typography openSans should render font 1`] = `
<div
  className="sc-AxjAm bYePpP"
  fontFamily="openSans"
  size="normal"
>
  Test
</div>
`;

exports[`Typography should render 1`] = `
<div
  className="sc-AxjAm htRihw"
  fontFamily={null}
  size="normal"
>
  Test
</div>
`;
